### PR TITLE
[CodeCompletion] Don't suggest underscored module name

### PR DESCRIPTION
**Pull request by rintaro**
Underscored modules (e.g. `_Concurrency`) are usually not meant be written explicitly in source code. They are usually implicitly imported with other modules. Code completion should not suggest them in type and expression context.

rdar://78232229

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@swift-ci Please smoke test Windows

File: test/IDE/complete_modulename.swift [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+// EXPR-DAG: Decl[Struct]/OtherModule[MyModule]: MyModuleTy[#MyModuleTy#]; name=MyModuleTy
+// EXPR-DAG: Decl[Struct]/OtherModule[_Hidden]:  HiddenTy[#HiddenTy#]; name=HiddenTy
+// EXPR-DAG: Decl[FreeFunction]/OtherModule[MyModule]: myModuleFunc()[#Void#]; name=myModuleFunc()
+// EXPR-DAG: Decl[FreeFunction]/OtherModule[_Hidden]: hiddenFunc()[#Void#]; name=hiddenFunc()
+// EXPR: End completions
+
+// TYPE: Begin completion
+// TYPE-NOT: _Concurrency[#Module#]
+// TYPE-NOT: SwiftShims[#Module#]
+// TYPE-NOT: SwiftOnoneSupport[#Module#]
```
Comment (benlangmuir) [inline test/IDE/complete_modulename.swift:37]: What about _Hidden? I actually think we should show it here since it is explicitly imported.

Comment (rintaro) [reply test/IDE/complete_modulename.swift:37]: Oh, I think I just forgot to add a line for `_Hidden`.
Anyway, let's make it show explicitly imported modules regardless.

Comment (rintaro) [reply test/IDE/complete_modulename.swift:50]: Added a commit. How does it look?

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Review by benlangmuir — APPROVED** [review]

**Comment by rintaro** [comment]
Windows failure are unrelated XPASS https://github.com/apple/swift/pull/37545
Merging

**Pull request merged**